### PR TITLE
Add a CommitResult type.

### PR DESCRIPTION
**Pull request by mr-salty**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp-spanner/269)

### CODE REVIEW

File: google/cloud/spanner/commit_result.h [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#include "google/cloud/spanner/timestamp.h"
+#include "google/cloud/spanner/version.h"
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+/**
+ * The result of Committing a Transaction.
```
Comment (coryan) [inline google/cloud/spanner/commit_result.h]: nit: s/Committing/committing/ ?

**Comment by mr-salty** [comment]
FYI, The design doc has this as a class, but it seems to me that a struct is sufficient unless we have some philosophical objection to it.

File: google/cloud/spanner/commit_result.h [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+struct CommitResult {
+ public:
```
Comment (devjgm) [inline google/cloud/spanner/commit_result.h]: We should probably omit the `public:` for structs.

Comment (mr-salty) [reply google/cloud/spanner/commit_result.h]: oops - I forgot to delete it when I made it a struct - thanks!

**Pull request merged**